proxy/ProxySession.h

```diff
@@ -0,0 +1,188 @@
+/** @file
```

Comment:
I noticed that git noticed that some of the files were renamed (e.g. ProxyClientSession.cc -> ProxySession.cc above.  But it isn't tracking that the .h moved.

I think git might have a prayer of being smart with future rebases and cherry picks if it knows the history through the name change.